packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -81,13 +81,16 @@ export function createInsightsMiddleware<
             window[pointer].queue.push(args);
           };
           window[pointer].version = ALGOLIA_INSIGHTS_VERSION;
+          window[pointer].shouldAddScript = true;
         }
 
-        insightsClient = window[pointer];
-        insightsClient.shouldAddScript = true;
```

Comment:
we should only add the script if we wrote window.aa, as otherwise nothing will happen with the script